src/mosh.ts

```diff
@@ -0,0 +1,124 @@
+import { knownHosts, configHosts } from "./ssh";
+
+const completionSpec: Fig.Spec = {
+  name: "mosh",
+  description: "",
+  args: {
+    name: "user@hostname",
+    description: "Address of remote machine to log into",
+    generators: [knownHosts, configHosts, { template: "history" }],
+  },
+  options: [
+    {
+      name: ["--help", "-h"],
+      description: "Show help for mosh",
+    },
+    {
+      name: "--client",
+      description: 'Mosh client on local machine (default: "mosh-client")',
+    },
+    {
+      name: "--server",
+      description: 'Mosh server on remote machine (default: "mosh-server")',
+    },
+    {
+      name: "--predict=adaptive",
+      description: "Local echo for slower links [default]",
+    },
+    {
+      name: ["--predict=always", "-a"],
+      description: "Use local echo even on fast links",
+    },
+    {
+      name: ["--predict=never", "-n"],
+      description: "Never use local echo",
+    },
+    {
+      name: "--predict=experimental",
+      description: "Aggressively echo even when incorrect",
+    },
+    {
+      name: "-4",
+      description: "Use IPv4 only",
+    },
+    {
+      name: "-6",
+      description: "Use IPv6 only",
+    },
+    {
+      name: "--family",
+      description: "Network Type",
```

Comment:
If an equals is required between the option and the argument you should add this option:
```suggestion
      description: "Network Type",
      requiresSeparator: true,
```